kroxylicious-filters/kroxylicious-record-encryption/etc/module-layering.xml

```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0"?>
+<!--
+
+    Copyright Kroxylicious Authors.
+
+    Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<!DOCTYPE import-control PUBLIC
+        "-//Checkstyle//DTD ImportControl Configuration 1.4//EN"
+        "https://checkstyle.org/dtds/import_control_1_4.dtd">
+
+<import-control pkg="io.kroxylicious.filter.encryption">
+    <!-- Allowed globally -->
+    <allow pkg="javax?\..*" regex="true"/>
+    <allow pkg="edu.umd.cs.findbugs.annotations"/>
+    <allow pkg="org.slf4j"/>
+    <allow pkg="io.micrometer.core.instrument"/>
+    <allow pkg="io.kroxylicious.proxy.tag"/>
+    <allow pkg="io.kroxylicious.filter.encryption.common"/>
+
+    <!-- top level package (these are all local-only=true)  -->
+    <allow pkg="org.apache.kafka.common.message" local-only="true"/>
```

Comment:
Trying to grok `local-only`. The DTD says `local-only - Indicates that the rule is to apply only to the current package and not to subpackages.`

So is this rule saying any class in `io.kroxylicious.filter.encryption` (or any subpackage) can import anything from `org.apache.kafka.common.message` but not from subpackages of `org.apache.kafka.common.message`?

Reply:
> local-only - Indicates that the rule is to apply only to the current package and not to subpackages.

I'd not noticed that that's ambiguous. Does "current package" mean the package matched by `pkg`, or the one named in the `subpackage` element? 

Experimentation confirms that it means the latter. So this rule means classes in `io.kroxylicious.filter.encryption` can import `org.apache.kafka.common.message`, but classes in a subpackage of `io.kroxylicious.filter.encryption` cannot (unless there's an `allow` for `org.apache.kafka.common.message` in the `subpackage` element for that subpackage).

So:
* lines 15-20 (without the `local-only`) are blanket rules saying "allow importing these into `io.kroxylicious.filter.encryption` *and any subpackage package*". 
* lines 23-28 (with the `local-only`) are saying "allow importing these into `io.kroxylicious.filter.encryption` _only_. 
* and the `subpackage` elements that follow contain specific rules for each subpackage of `io.kroxylicious.filter.encryption`.